src/toolkit/matl_buy_policy.cc

```diff
@@ -43,18 +45,46 @@ void MatlBuyPolicy::set_manager(Agent* m) {
   }
 }
 
+void MatlBuyPolicy::set_total_inv_tracker(TotalInvTracker* t) {
+  if (t == NULL){
+    std::vector<ResBuf<Material>*> bufs = {buf_};
+    buf_tracker_->Init(bufs, buf_->capacity());
+  }
+  else {
+    buf_tracker_ = t;
```

Comment:
should we make sure that `buf_` is part of `buf_tracker_`?  That is, make sure that the `buf_tracker_` includes `buf_` as one of it's buffers?